entity-view/impl/src/main/java/com/blazebit/persistence/view/impl/EntityViewManagerImpl.java

```diff
@@ -244,9 +244,56 @@ public ProxyFactory getProxyFactory() {
     public <T> T find(EntityManager entityManager, EntityViewSetting<T, CriteriaBuilder<T>> entityViewSetting, Object entityId) {
         ViewTypeImpl<T> managedViewType = metamodel.view(entityViewSetting.getEntityViewClass());
         EntityType<?> entityType = (EntityType<?>) managedViewType.getJpaManagedType();
-        SingularAttribute<?, ?> idAttribute = JpaMetamodelUtils.getSingleIdAttribute(entityType);
-        CriteriaBuilder<?> cb = cbf.create(entityManager, managedViewType.getEntityClass())
-                .where(idAttribute.getName()).eq(entityId);
+        Set<SingularAttribute<?, ?>> idAttributeSet = JpaMetamodelUtils.getIdAttributes(entityType);
+        CriteriaBuilder<?> cb = cbf.create(entityManager, managedViewType.getEntityClass());
+        Map<String,Object> entityIdValues = new HashMap<>();
+        //TODO Hibernate returns null when getIdType it is called on an entityType which uses IdClass.
+        if(entityType.getIdType()!=null && (entityType.getIdType().getPersistenceType().equals(Type.PersistenceType.BASIC)||
+                entityType.getIdType().getPersistenceType().equals(Type.PersistenceType.EMBEDDABLE))){
+            Iterator iterator = idAttributeSet.iterator();
+            if(!iterator.hasNext()){
+                throw new RuntimeException("The entity type" + entityType.getName() + "does not have an Id!");
+                }
+
+            //I believe this approach also works for BigDecimal and the like? I have added that possibility in the check above.
+            entityIdValues.put(((SingularAttribute) iterator.next()).getName(),entityId);
+
+            if (iterator.hasNext()){
+                throw new RuntimeException("Could not match the given entityId to the entity type specified in the view: the entity type"
+                        + entityType.getName() + "has more than one Id field!");
+            }
+        } else {
+//            for (Field field : Arrays.asList(entityId.getClass().getDeclaredFields())){
+//                field.setAccessible(Boolean.TRUE);
```

Comment:
I don't think we need this anymore, finding the field based on the provided member should probably work. If not, we can still lookup the field by attribute name, as these, according to the spec, should match as far as I remember.